langchain/vectorstores/qdrant.py

```diff
@@ -462,17 +462,37 @@ def _document_from_scored_point(
         )
 
     def _qdrant_filter_from_dict(self, filter: Optional[MetadataFilter]) -> Any:
-        if filter is None or 0 == len(filter):
+        if not filter:
             return None
 
         from qdrant_client.http import models as rest
 
+        def _build_condition(key: str, value: Any) -> List[rest.FieldCondition]:
```

Comment:
(if it's just about type hinting with optional imports there's other ways to get around that)

Reply:
(if it's just about type hinting with optional imports there's other ways to get around that)

Reply:
Oh, it's not strictly needed to declare this within `_qdrant_filter_from_dict`. I thought it would clutter things if I declared it a separate private class method. XD 

Does this look better?

```Python
    def _build_condition(self, key: str, value: Any) -> List[Any]:
        from qdrant_client.http import models as rest

        out = []

        if isinstance(value, dict):
            for _key, value in value.items():
                out.extend(self._build_condition(f"{key}.{_key}", value))
        elif isinstance(value, list):
            for _value in value:
                if isinstance(_value, dict):
                    out.extend(self._build_condition(f"{key}[]", _value))
                else:
                    out.extend(self._build_condition(f"{key}", _value))
        else:
            out.append(
                rest.FieldCondition(
                    key=f"{self.metadata_payload_key}.{key}",
                    match=rest.MatchValue(value=value),
                )
            )

        return out

    def _qdrant_filter_from_dict(self, filter: Optional[MetadataFilter]) -> Any:
        if not filter:
            return None

        from qdrant_client.http import models as rest

        return rest.Filter(
            must=[
                condition
                for key, value in filter.items()
                for condition in self._build_condition(key, value)
            ]
        )
```

> (if it's just about type hinting with optional imports there's other ways to get around that)

I had to do a separate import for the two methods + use `Any`, but if you have other suggestions, please let me know! Then I'll update the PR. Thanks! :)

Reply:
looks great, i like it split out! personally am fine with double import

for typing hinting you can actually refer to rest (rest.Filter or List[rest.FieldCondition]) as long as you add at the top of the file
```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
  from qdrant_client.http import models as rest
```